scripts/gen_relocate_app.py

```diff
@@ -0,0 +1,370 @@
+#!/usr/bin/env python3
```

Comment:
Please spend some time on this script and fix the syntax, you get it to a higher score, at least 8.00 :)

```
************* Module gen_relocate_app
scripts/gen_relocate_app.py:12:0: C0301: Line too long (110/100) (line-too-long)
scripts/gen_relocate_app.py:75:29: C0326: Exactly one space required before assignment
extern_linker_var_declaration= """
                             ^ (bad-whitespace)
scripts/gen_relocate_app.py:101:15: C0326: Exactly one space required before assignment
memset_template= """
               ^ (bad-whitespace)
scripts/gen_relocate_app.py:108:26: C0326: No space allowed after bracket
        full_lib = ELFFile( f)
                          ^ (bad-whitespace)
scripts/gen_relocate_app.py:109:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
scripts/gen_relocate_app.py:110:40: C0326: Exactly one space required after comma
            print("Error parsing file: ",filename)
                                        ^ (bad-whitespace)
scripts/gen_relocate_app.py:113:19: C0326: No space allowed after bracket
        sections = [ x for x in full_lib.iter_sections()]
                   ^ (bad-whitespace)
scripts/gen_relocate_app.py:118:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
scripts/gen_relocate_app.py:121:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
scripts/gen_relocate_app.py:124:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
scripts/gen_relocate_app.py:127:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
scripts/gen_relocate_app.py:135:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
scripts/gen_relocate_app.py:136:26: C0326: No space allowed after bracket
                symbols = [ x for x in section.iter_symbols()]
                          ^ (bad-whitespace)
scripts/gen_relocate_app.py:139:0: C0301: Line too long (120/100) (line-too-long)
scripts/gen_relocate_app.py:150:49: C0326: Exactly one space required after comma
        memory_type = memory_type.replace("_TEXT","")
                                                 ^ (bad-whitespace)
scripts/gen_relocate_app.py:153:51: C0326: Exactly one space required after comma
        memory_type = memory_type.replace("_RODATA","")
                                                   ^ (bad-whitespace)
scripts/gen_relocate_app.py:156:49: C0326: Exactly one space required after comma
        memory_type = memory_type.replace("_DATA","")
                                                 ^ (bad-whitespace)
scripts/gen_relocate_app.py:159:48: C0326: Exactly one space required after comma
        memory_type = memory_type.replace("_BSS","")
                                                ^ (bad-whitespace)
scripts/gen_relocate_app.py:167:0: C0330: Wrong continued indentation before block (add 4 spaces).
                full_list_of_sections[iter_sec] != []):
                ^   | (bad-continuation)
scripts/gen_relocate_app.py:173:64: C0326: No space allowed before bracket
        tmp_list = {"text":[], "rodata":[], "data":[], "bss":[] }
                                                                ^ (bad-whitespace)
scripts/gen_relocate_app.py:176:0: C0330: Wrong continued indentation before block (add 4 spaces).
                full_list_of_sections[iter_sec] != []):
                ^   | (bad-continuation)
scripts/gen_relocate_app.py:197:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
scripts/gen_relocate_app.py:229:21: C0326: Exactly one space required after assignment
    generate_section ={"text":False, "rodata":False, "data":False, "bss":False }
                     ^ (bad-whitespace)
scripts/gen_relocate_app.py:229:79: C0326: No space allowed before bracket
    generate_section ={"text":False, "rodata":False, "data":False, "bss":False }
                                                                               ^ (bad-whitespace)
scripts/gen_relocate_app.py:233:58: C0326: Exactly one space required after comma
            memory_type = memory_type.replace(section_name,"")
                                                          ^ (bad-whitespace)
scripts/gen_relocate_app.py:244:24: C0326: Exactly one space required after comma
    for mtype in ["text","rodata", "data"]:
                        ^ (bad-whitespace)
scripts/gen_relocate_app.py:263:0: W0301: Unnecessary semicolon (unnecessary-semicolon)
scripts/gen_relocate_app.py:287:0: C0330: Wrong continued indentation (add 12 spaces).
            help="input src:memory type(sram2 or ccm or aon etc) string")
            ^           | (bad-continuation)
scripts/gen_relocate_app.py:291:67: C0326: No space allowed before keyword argument assignment
    parser.add_argument("-v", "--verbose", action="count", default =0,
                                                                   ^ (bad-whitespace)
scripts/gen_relocate_app.py:304:51: C0326: Exactly one space required after comma
                    fullname = os.path.join(dirpath,filename1)
                                                   ^ (bad-whitespace)
scripts/gen_relocate_app.py:325:46: C0326: Exactly one space required after comma
            print("Memory region ", mem_region," Selected for file:", file_name_list)
                                              ^ (bad-whitespace)
scripts/gen_relocate_app.py:331:0: C0325: Unnecessary parens after 'return' keyword (superfluous-parens)
scripts/gen_relocate_app.py:346:20: C0326: Exactly one space required after comma
    for  memory_type,files in rel_dict.items():
                    ^ (bad-whitespace)
scripts/gen_relocate_app.py:347:30: C0326: Exactly one space required after assignment
        full_list_of_sections ={"text":[], "rodata":[], "data":[], "bss":[] }
                              ^ (bad-whitespace)
scripts/gen_relocate_app.py:347:76: C0326: No space allowed before bracket
        full_list_of_sections ={"text":[], "rodata":[], "data":[], "bss":[] }
                                                                            ^ (bad-whitespace)
scripts/gen_relocate_app.py:1:0: C0111: Missing module docstring (missing-docstring)
scripts/gen_relocate_app.py:37:0: C0103: Constant name "print_template" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/gen_relocate_app.py:41:0: C0103: Constant name "group_start_end_memory_seq" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/gen_relocate_app.py:45:0: C0103: Constant name "section_load_memory_seq" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/gen_relocate_app.py:49:0: C0103: Constant name "load_address_location_flash" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/gen_relocate_app.py:50:0: C0103: Constant name "load_address_location_bss" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/gen_relocate_app.py:53:0: C0103: Constant name "linker_section_seq" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/gen_relocate_app.py:68:0: C0103: Constant name "source_code_includes" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/gen_relocate_app.py:75:0: C0103: Constant name "extern_linker_var_declaration" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/gen_relocate_app.py:82:0: C0103: Constant name "data_copy_function" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/gen_relocate_app.py:89:0: C0103: Constant name "bss_zeroing_function" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/gen_relocate_app.py:96:0: C0103: Constant name "memcpy_template" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/gen_relocate_app.py:101:0: C0103: Constant name "memset_template" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/gen_relocate_app.py:106:0: C0111: Missing function docstring (missing-docstring)
scripts/gen_relocate_app.py:107:33: C0103: Variable name "f" doesn't conform to snake_case naming style (invalid-name)
scripts/gen_relocate_app.py:111:12: E1101: Module 'os' has no 'exit' member; maybe '_exit'? (no-member)
scripts/gen_relocate_app.py:144:0: C0111: Missing function docstring (missing-docstring)
scripts/gen_relocate_app.py:184:0: C0111: Missing function docstring (missing-docstring)
scripts/gen_relocate_app.py:185:4: W0621: Redefining name 'string' from outer scope (line 29) (redefined-outer-name)
scripts/gen_relocate_app.py:190:0: C0111: Missing function docstring (missing-docstring)
scripts/gen_relocate_app.py:192:4: W0621: Redefining name 'string' from outer scope (line 29) (redefined-outer-name)
scripts/gen_relocate_app.py:214:0: C0111: Missing function docstring (missing-docstring)
scripts/gen_relocate_app.py:215:4: W0621: Redefining name 'string' from outer scope (line 29) (redefined-outer-name)
scripts/gen_relocate_app.py:223:36: C0103: Variable name "fw" doesn't conform to snake_case naming style (invalid-name)
scripts/gen_relocate_app.py:226:0: C0111: Missing function docstring (missing-docstring)
scripts/gen_relocate_app.py:258:0: C0111: Missing function docstring (missing-docstring)
scripts/gen_relocate_app.py:259:4: W0621: Redefining name 'string' from outer scope (line 29) (redefined-outer-name)
scripts/gen_relocate_app.py:275:35: C0103: Variable name "fw" doesn't conform to snake_case naming style (invalid-name)
scripts/gen_relocate_app.py:279:0: C0111: Missing function docstring (missing-docstring)
scripts/gen_relocate_app.py:280:4: C0103: Constant name "args" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/gen_relocate_app.py:280:4: W0601: Global variable 'args' undefined at the module level (global-variable-undefined)
scripts/gen_relocate_app.py:296:0: C0111: Missing function docstring (missing-docstring)
scripts/gen_relocate_app.py:296:0: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
scripts/gen_relocate_app.py:300:17: W0612: Unused variable 'dirs' (unused-variable)
scripts/gen_relocate_app.py:309:0: C0111: Missing function docstring (missing-docstring)
scripts/gen_relocate_app.py:314:8: E1101: Module 'os' has no 'exit' member; maybe '_exit'? (no-member)
scripts/gen_relocate_app.py:334:0: C0111: Missing function docstring (missing-docstring)
scripts/gen_relocate_app.py:342:4: W0612: Unused variable 'raw' (unused-variable)
scripts/gen_relocate_app.py:25:0: W0611: Unused import sys (unused-import)
scripts/gen_relocate_app.py:28:0: W0611: Unused import re (unused-import)
scripts/gen_relocate_app.py:29:0: W0611: Unused import string (unused-import)
scripts/gen_relocate_app.py:31:0: C0411: standard import "import glob" should be placed before "from elftools.elf.elffile import ELFFile" (wrong-import-order)
scripts/gen_relocate_app.py:32:0: C0411: standard import "import warnings" should be placed before "from elftools.elf.elffile import ELFFile" (wrong-import-order)

-----------------------------------
Your code has been rated at 5.35/10
```

Reply:
pylint

Reply:
@nashif comments handled. please re-review.